src/analysis/ConfigurationsExt.cpp

```diff
@@ -304,6 +306,209 @@ namespace espressopp {
       delete [] ids;
     }
 
+    void ConfigurationsExt::gather_modified() {
+
+          System& system = getSystemRef();
+
+          // determine number of local particles and total particles
+
+          int myN = system.storage->getNRealParticles();
+          int maxN;   // maximal number of particles one processor has
+          int totalN; // total number of particles all processors have
+
+          boost::mpi::all_reduce(*system.comm, myN, maxN, boost::mpi::maximum<int>());
+          boost::mpi::all_reduce(*system.comm, myN, totalN, std::plus<int>());  // probably there is an equivalent in boost:mpi?DN
+
+          LOG4ESPP_INFO(logger, "#Partices: me = " << myN << ", max = " << maxN
+                                << ", totalN = " << totalN);
+
+          real* coordinates = new real [3 * maxN];  // buffer for gather
+          real* velocities  = new real [3 * maxN];  // buffer for gather
+          int*  ids         = new int [maxN];  // buffer for gather
+
+          // add types
+          size_t* types = new size_t [maxN];  // buffer for gather
+
+          // fill the buffer with my values
+
+          CellList realCells = system.storage->getRealCells();
+
+          int i = 0;
+
+          if( unfolded ){
+            for(CellListIterator cit(realCells); !cit.isDone(); ++cit) {
+
+              ids[i] = cit->id();
+              types[i] = cit->type(); // type of particle added
+
+              Real3D& pos = cit->position();
+              Real3D& vel = cit->velocity();
+              Int3D& img = cit->image();
+              Real3D L = system.bc->getBoxL();
+
+              coordinates[3*i]   = pos[0] + img[0] * L[0];
+              coordinates[3*i+1] = pos[1] + img[1] * L[1];
+              coordinates[3*i+2] = pos[2] + img[2] * L[2];
+
+              velocities[3*i]   = vel[0];
+              velocities[3*i+1] = vel[1];
+              velocities[3*i+2] = vel[2];
+
+              i++;
+            }
+          }
+          else{
+            for(CellListIterator cit(realCells); !cit.isDone(); ++cit) {
+
+              ids[i] = cit->id();
+              types[i] = cit->type(); // type of particle added
+
+              Real3D& pos = cit->position();
+              Real3D& vel = cit->velocity();
+
+              coordinates[3*i]   = pos[0];
+              coordinates[3*i+1] = pos[1];
+              coordinates[3*i+2] = pos[2];
+
+              velocities[3*i]   = vel[0];
+              velocities[3*i+1] = vel[1];
+              velocities[3*i+2] = vel[2];
+
+              i++;
+            }
+          }
+
+          if (i != myN) {
+            LOG4ESPP_ERROR(logger, "mismatch for number of local particles");
+          }
+
+          // each proc sends its data to master process
+
+          if (system.comm->rank() == 0) {
```

Comment:
That is the logic. And since I don't know if there are really performance improvement I wouldn't suggest to modify it. But we can think about that.